packages/composite-checkout/src/components/checkout-step.js

```diff
@@ -118,12 +118,12 @@ CheckoutStepHeader.propTypes = {
 	onEdit: PropTypes.func,
 };
 
-function Stepper( { isComplete, isActive, className, children } ) {
+function Stepper( { isComplete, isActive, className, children, stepNumber } ) {
 	return (
 		<StepNumberOuterWrapper className={ joinClasses( [ className, 'checkout-step__stepper' ] ) }>
 			<StepNumberInnerWrapper isComplete={ isComplete }>
 				<StepNumber isComplete={ isComplete } isActive={ isActive }>
-					{ children }
+					{ stepNumber !== 0 && children }
```

Comment:
Because I'm also touching the step number and making it optional in #37728 I'm going to wait on approving this PR.

Reply:
ok, no problem. I'll wait for that PR to merge and update this one.

Reply:
It's not necessary anymore, this code was removed.